parcels/particle.py

```diff
@@ -33,6 +33,34 @@ def AdvectionEE(particle, grid, time, dt):
     particle.lat += v1 * f_lat
 
 
+def positions_from_density_field(pnum, startfield, mode='monte_carlo'):
+    # initialise particles from a field
+    print("Starting from " + startfield.name + " field")
+
+    # Gradient ends up with overall negative sum of cells, not sure if this makes sense
+    # so working with lat long points still for the moment
+    # startcells = np.gradient(startfield.data[0,:,:])
+    # total = np.sum(startcells)
```

Comment:
No- these are a commented out attempts to move from assigning particle positions at lat-lon points to assigning them to cells (you remember we discussed using np.gradient to do this last week?). But it doesn't work out as I expected because of course we end up with some negative gradient values assigned to some cells, which is not what we wanted I think.

The print statement can be removed if you want, I thought it was just a useful line out for the user.